contrib/genbuild.sh

```diff
@@ -25,7 +25,7 @@ git_check_in_repo() {
 DESC=""
 SUFFIX=""
 CURRENT_BRANCH=""
-if [ -z "$BUILD_VERSION" ] && [ "${BITCOIN_GENBUILD_NO_GIT}" != "1" ] && [ -e "$(command -v git)" ] && [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" = "true" ] && git_check_in_repo contrib/genbuild.sh; then
+if [ "${BITCOIN_GENBUILD_NO_GIT}" != "1" ] && [ -e "$(command -v git)" ] && [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" = "true" ] && git_check_in_repo contrib/genbuild.sh; then
```

Comment:
Is there a specific reason to remove this?

Reply:
Skipping the dirty tag check when BUILD_VERSION is specified is incorrect in our previous docker build workflow because BUILD_VERSION was not passed on into the docker image itself which resulted in the defid version still being tagged as dirty. This was removed and reverted because with CI on local builds inside the runner, build should be clean and there is no need to explicitly skip the check.

In addition, BUILD_VERSION specified in CI will vary depending on if it is a tagged commit. Thus on build-release workflow, it should be passed directly as BUILD_DESC, while on merge to master, BUILD_VERSION should be defined as BUILD_SUFFIX. It would be easier to simply revert back to the old genbuild pipeline, which will resolve the incorrect dirty tag once we shift our build locally.